patches/tModLoader/Terraria/Audio/SoundStyle.TML.cs

```diff
@@ -58,6 +59,14 @@ public record struct SoundStyle
 	/// <summary> Determines what the action taken when the max amount of sound instances is reached. </summary>
 	public SoundLimitBehavior SoundLimitBehavior { get; set; } = SoundLimitBehavior.ReplaceOldest;
 
+	/// <summary>
+	/// How many times to attempt to find a variant that is not currently playing before applying the SoundLimitBehavior. Only has effect if LimitsArePerVariant is true. Defaults to 1.
+	/// </summary>
+	public int RerollAttempts {
```

Comment:
Since it's called _Re_roll attempts, it should default to 0, and thunder rerolls 6 times